incubator/hnc/internal/reconcilers/object.go

```diff
@@ -666,7 +665,10 @@ func (r *ObjectReconciler) shouldPropagateSource(log logr.Logger, inst *unstruct
 	if err != nil {
 		log.Error(err, "Invalid value")
 	}
-	noneSelector := r.getNoneSelector(log, inst)
+	noneSelector, err := selectors.GetNoneSelector(inst)
+	if err != nil {
```

Comment:
Hmm. I feel like there's a logic bug for the other two selectors here: if they return an error, then this object shouldn't be propagated at all, right? But they're also returning `nil` selectors so we're just going to ignore them, right?

Should this function just `return false` if any of the selectors have an error? Do we have a test for this?

Reply:
I agree. We do have tests for this, but at that time I don't have a clear understanding of a "bad selector" so I didn't choose the value properly. I updated the logic and the test

Reply:
Did you update this in https://github.com/kubernetes-sigs/multi-tenancy/pull/1225? I'm not seeing the update?